core/src/main/java/org/mule/runtime/core/DefaultMessageExecutionContext.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.core;
+
+import static java.lang.String.format;
+import static java.time.OffsetTime.now;
+import static java.util.Optional.ofNullable;
+
+import org.mule.runtime.core.api.MessageExecutionContext;
+import org.mule.runtime.core.api.MuleContext;
+import org.mule.runtime.core.api.MuleEvent;
+import org.mule.runtime.core.api.config.MuleConfiguration;
+import org.mule.runtime.core.api.construct.FlowConstruct;
+import org.mule.runtime.core.api.source.MessageSource;
+
+import java.io.Serializable;
+import java.time.OffsetTime;
+import java.util.Optional;
+
+/**
+ * Default immutable implementation of {@link MessageExecutionContext}.
+ *
+ * @since 4.0
+ */
+public final class DefaultMessageExecutionContext implements MessageExecutionContext, Serializable {
+
+  private static final long serialVersionUID = -3664490832964509653L;
+
+  /**
+   * Builds a new execution context with the given parameters.
+   * @param flow the flow that processes events of this context. 
+   */
+  public static MessageExecutionContext createContext(FlowConstruct flow) {
```

Comment:
Just "Create"? Because this can't be used with a static import as it is because it's not clear if it creates a MuleContext or ExceutionContext etc.